properties/utils.py

```diff
@@ -147,20 +147,20 @@ class ValidationError(ValueError):
     def __init__(self, message, reason=None, prop=None, instance=None,
                  _error_tuples=None):
         super(ValidationError, self).__init__(message)
-        if _error_tuples is None:
-            self.error_tuples = []
-        else:
-            self.error_tuples = _error_tuples
-        if reason is not None and not isinstance(reason, string_types):
+        if reason and not isinstance(reason, string_types):
             raise TypeError('ValidationError reason must be a string')
-        if prop is not None and not isinstance(prop, string_types):
+        if prop and not isinstance(prop, string_types):
             raise TypeError('ValidationError prop must be a string')
-        if instance is not None and not hasattr(instance, '_error_hook'):
+        if instance and not hasattr(instance, '_error_hook'):
             raise TypeError('ValidationError instance must be a '
                             'HasProperties instance')
-        if reason or prop or instance:
-            error_tuple = ErrorTuple(message, reason, prop, instance)
-            self.error_tuples.append(error_tuple)
+        error_tuple = ErrorTuple(message, reason, prop, instance)
+        if _error_tuples is None:
+            self.error_tuples = [error_tuple]
+        else:
+            self.error_tuples = _error_tuples
+            if reason or prop or instance:
+                self.error_tuples.append(error_tuple)
```

Comment:
Is there any case where we care about validators passing things other than `None` for `reason`, `prop` or `instance`?

This passes:
```python

    def test_invalid_error_tuple_info(self):

        class RaisesError(properties.HasProperties):

            @properties.validator
            def raise_error(self):
                raise properties.ValidationError('', False, False, False)

        errorer = RaisesError()
        with self.assertRaises(properties.ValidationError):
            errorer.validate()
```

(i.e., they get past the `TypeError` checks)

I didn't quite follow the reasoning for those to be `None` above; could we tighten them down by defaulting `''` instead of `None`? It is also a singleton.

Reply:
Cool, totally makes sense.

Reply:
On second look - I left the `None` defaults. Changing to `''` and validating against that would introduce some unnecessary backwards incompatibility. I did tighten up the errors to prevent introducing other things that evaluate to `False`, though.

Reply:
:thumbsup: Works for me